### PR TITLE
fix: preview mode glitch

### LINKED ISSUE #30542: [Bug]: Preview mode transition bugs

### PR DESCRIPTION
**Pull request by AmanAgarwal041**
Fixes #30542
_or_  
Fixes `Issue URL`
> [!WARNING]  
> _If no issue exists, please create an issue first, and check with the maintainers if the issue is valid._

## Automation

/ok-to-test tags=""

### :mag: Cypress test results

> [!CAUTION]  
> If you modify the content in this section, you are likely to disrupt the CI result for your PR.

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [ ] No

## Summary by CodeRabbit

- **Improvements**
  - Updated canvas component animation to be instantaneous for a more responsive experience.
  - Enhanced IDE component to dynamically adjust margins based on preview mode, ensuring better layout adaptability.

### CODE REVIEW

File: app/client/src/pages/Editor/IDE/index.tsx [hunk 1]
```diff
@@ -24,29 +24,46 @@ function IDE() {
...
       {isProtectedMode && <ProtectedCallout />}
       <EditorWrapperContainer>
         <div
           className={classNames({
-            [`transition-transform transform duration-400 flex ${tailwindLayers.entityExplorer}`]:
+            [`transition-all transform duration-400 flex ${tailwindLayers.entityExplorer}`]:
               true,
-            relative: !isCombinedPreviewMode,
-            "-translate-x-full fixed": isCombinedPreviewMode,
+            // relative: !isCombinedPreviewMode,
```
Comment (jsartisan) [inline app/client/src/pages/Editor/IDE/index.tsx:50]: Can we remove this comment if not required?

File: app/client/src/pages/Editor/Canvas.tsx [hunk 2]
```diff
@@ -94,7 +94,7 @@ const Canvas = (props: CanvasProps) => {
         <Wrapper
           $enableMainCanvasResizer={!!props.enableMainCanvasResizer}
           background={isAnvilLayout ? "" : backgroundForCanvas}
-          className={`relative t--canvas-artboard ${paddingBottomClass} transition-all duration-400  ${marginHorizontalClass} ${getViewportClassName(
+          className={`relative t--canvas-artboard ${paddingBottomClass} transition-none  ${marginHorizontalClass} ${getViewportClassName(
```
Comment (jsartisan) [inline app/client/src/pages/Editor/Canvas.tsx:97]: Are we removing animation here?

File: app/client/src/pages/Editor/IDE/index.tsx [hunk 3]
```diff
@@ -24,29 +24,46 @@ function IDE() {
...
+  const leftRef = useRef<HTMLDivElement>(null);
+  const rightRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    if (leftRef.current && rightRef.current) {
+      if (isCombinedPreviewMode) {
+        leftRef.current.style.marginRight = `-${leftRef.current.getBoundingClientRect().width}px`;
+        rightRef.current.style.marginLeft = `-${rightRef.current.getBoundingClientRect().width}px`;
+      } else {
+        leftRef.current.style.marginRight = "0px";
```
Comment (jsartisan) [inline app/client/src/pages/Editor/IDE/index.tsx:36]: Why are we doing this? Can't we do this with CSS? 🤔

File: app/client/src/pages/Editor/Canvas.tsx (hunk 2 above)
Comment (AmanAgarwal041) [reply app/client/src/pages/Editor/Canvas.tsx:97]: This was creating a janky effect and after removing this it became a little smoother.

File: app/client/src/pages/Editor/IDE/index.tsx (hunk 3 above)
Comment (AmanAgarwal041) [reply app/client/src/pages/Editor/IDE/index.tsx:36]: Since the property pane is resizeable, hence it would be better to add a margin using the js.

**Comment by AmanAgarwal041** [comment]
Closing this PR due to  https://github.com/appsmithorg/appsmith/issues/30542#issuecomment-2177963754

**Pull request closed without merging**